### ISSUE TITLE
Whitelist afschaffen

### ISSUE THREAD

**Issue by DanielVanHilten** [TweedeKamerDerStaten-Generaal/OpenDataAPI2.0#1]
Vele afnemers hebben aangegeven de IP whitelist omslachtig te vinden en een te hoge drempel is. Intern wordt er onderzocht of we dit uit kunnen gaan zetten.

**Comment by jurrian** [comment]
Is er ondertussen al meer bekend over wanneer de whitelist eventueel zal worden afgeschaft?

**Comment by DanielVanHilten** [comment]
Inmiddels liggen de plannen om naar een Cloud oplossing te gaan klaar en is er een Proof of Concept in ontwikkeling dat alleen het OData punt in de cloud aanbiedt. Dit moet nog wel getoetst en geëvalueerd worden. Zodra hier meer over bekend is zullen we dit in een nieuwsartikel publiceren op de site.

**Comment by webpatser** [comment]
Dat zou fijn zijn. Inmiddels werken mijn twee IP nummers niet meer, en ik krijg geen reactie terug van mijn meerdere contact verzoeken ;-)

**Comment by DanielVanHilten** [comment]
Update: Dienst Automatisering gaat dit in de komende weken analyseren en onderzoeken of hier een andere oplossing voor kan komen. Parallel loopt het de Cloud oplossing ook nog.

**Comment by bartromgens** [comment]
@DanielVanHilten Dank voor de update.

Op dit moment ben ik afhankelijk van mijn ISP die regelmatig mijn IP wijzigt. Helaas wijzigt mijn IP op dit moment sneller dan mijn wijzigingen worden doorgevoerd. Daardoor kan ik al enkele weken geen gebruik meer maken van de TK API. Helaas kan ik daardoor ook geen tijd inplannen om ermee te werken.

Het is nu 6 dagen gelden dat ik een wijziging heb doorgegeven. Daar ik heb ik een email bevestiging van gehad. Deze is nog niet doorgevoerd. 

Hiermee wil ik vooral duidelijk maken hoe het gebruikers beïnvloedt. Het is niet mijn bedoeling om via dit kanaal te vragen om mijn IP door te voeren (al zou dat ook mooi zijn :) ). 

Is het misschien mogelijk om als tussenoplossing IP wijzigingen van 'geverifieerde' accounts automatische door te voeren? Ik snap dat dit binnen de huidige omgeving mogelijk teveel ontwikkelwerk is, maar overweeg het wel. 

Ik vind het (totdat er een structurele oplossing is) geen probleem om bijv. een dag te wachten, maar het is niet werkbaar zolang het niet duidelijk is hoelang het duurt voordat een IP wijziging wordt doorgevoerd. Dat kan in mijn ervaring 1 dag, een week, of pas na herinnering zijn.

**Comment by webpatser** [comment]
Inderdaad...  het duurt nu te lang. Ik wacht al maanden inmiddels. Meerdere malen gewijzigd met een e-mail bevestiging dat het aangepast zou worden. Hier in de comments het melden lijkt ook niet te werken.

Hopelijk kunnen jullie dat handmatige proces van IPS toevoegen aan de firewall versnellen of automatiseren?

**Comment by DanielVanHilten** [comment]
Het is inderdaad een sub-optimaal proces waar wij snel vanaf willen. Mocht het langer duren dan 2 a 3 werkdagen dan kun je via het supportformulier dit melden, vermeld svp je naam en IP adres die je op de whitelist wilt laten controleren. Het vaker muteren van IP-adressen betekent hier dat je vorige er tevens weer afgehaald kan worden, als je dit dus vaker doorvoert dan wordt de kans dat dit fout gaat hoger. Het handmatige proces is inderdaad onwenselijk maar helaas is dit wat we op dit moment kunnen bieden. Jullie opmerkingen en bevindingen helpen ons enorm bij het belang aan te geven om een betere oplossing te vinden voor de whitelisting, bedankt hiervoor!

**Comment by Jonakemon** [comment]
@DanielVanHilten Is er een update over het afschaffen van de whitelisting? Als ik naar de comments hierboven kijk is dat iets wat erg gewenst is.

**Comment by Jonakemon** [comment]
@DanielVanHilten Ik vroeg me af of je een update hebt over het afschaffen van de whitelisting.

**Comment by JPBenard** [comment]
We hebben zojuist een nieuw bericht op de website geplaatst: https://opendata.tweedekamer.nl/nieuws/stand-van-zaken-registratie-en-ip-whitelisting
Hoewel we er hard aan werken, zijn er helaas nog geen concrete vorderingen te melden voor de afnemers. Ons doel is echter nog steeds om van het proces van whitelisting af te komen.

**Comment by Jonakemon** [comment]
> We hebben zojuist een nieuw bericht op de website geplaatst: https://opendata.tweedekamer.nl/nieuws/stand-van-zaken-registratie-en-ip-whitelisting
> Hoewel we er hard aan werken, zijn er helaas nog geen concrete vorderingen te melden voor de afnemers. Ons doel is echter nog steeds om van het proces van whitelisting af te komen.

Thanks voor de reactie! Is er ergens een roadmap beschikbaar voor ons om onze eigen ontwikkeling op af te stemmen m.b.t. de whitelisting en andere aanpassingen?

Voor ons is het in productie gebruiken van de API te gevaarlijk vanwege het (relatief) hoge percentage downtime en de IP whitelisting :/

**Comment by JPBenard** [comment]
Hi, het is voor het whitelistproces op dit moment lastig om dat aan te geven, maar ik ga het voor je navragen bij de projectleider. Andere wijzigingen proberen zo goed mogelijk te commuiniceren op de website, maar alle suggesties voor verbetering zijn welkom :)

Voor wat betreft het hoge percentage downtime: kun je misschien aangeven op welke momenten jij downtime constateert? Wij hebben dat namelijk zelf niet kunnen constateren. Als we exacte tijdstippen weten waarop je er last van hebt, dan kunnen we het proberen te reproduceren. Alvast bedankt!

**Comment by Jonakemon** [comment]
Hi @JPBenard: ik zat fout m.b.t. de hoge percentage downtime. Ik baseerde me op nieuwsberichten uit 2019. Daar had ik zorgvuldiger naar moeten kijken. Ik heb m'n comment aangepast!

**Comment by Jonakemon** [comment]
FYI, ik heb een reverse proxy opgezet voor mensen die niet aan whitelisting kunnen / willen doen: 37.97.129.67. Bijvoorbeeld http://37.97.129.67/OData/v4/2.0/ is het endpoint voor OData. Geen garanties qua uptime o.i.d. Credits aan [Wepublic](https://wepublic.nl) voor het betalen.
